modules/appgw/variables.tf

```diff
@@ -1,169 +1,632 @@
+                            that evaluates the truthfulness of the condition.
+        - `ignore_case`   - (`string`, optional, defaults to `false`) Perform a case in-sensitive comparison.
+        - `negate`        - (`bool`, optional, defaults to `false`) Negate the result of the condition evaluation.
+      - `request_headers` - (`map`, optional) Map of request header, where header name is the key,
+                            header value is the value of the object in the map.
+      - `response_headers`- (`map`, optional) Map of response header, where header name is the key,
+                            header value is the value of the object in the map.
   EOF
-  default     = "TLSv1_2"
-  type        = string
+  type = map(object({
+    name = string
+    rules = optional(map(object({
+      name     = string
+      sequence = number
+      conditions = optional(map(object({
+        pattern     = string
+        ignore_case = optional(bool, false)
+        negate      = optional(bool, false)
+      })), {})
+      request_headers  = optional(map(string), {})
+      response_headers = optional(map(string), {})
+    })))
+  }))
+  validation {
+    condition     = length(flatten([for _, rewrite in var.rewrites : rewrite.name])) == length(distinct(flatten([for _, rewrite in var.rewrites : rewrite.name])))
+    error_message = "The `name` property has to be unique among all rewrites."
+  }
 }
 
-variable "ssl_policy_cipher_suites" {
+variable "rules" {
   description = <<-EOF
-  A list of accepted cipher suites. Required only for `ssl_policy_type` set to `Custom`. 
-  For possible values see [documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/application_gateway#cipher_suites).
+  A map of rules for the Application Gateway.
+
+  A rule combines, http settings and health check configuration.
+  A key is an application name that is used to prefix all components inside Application Gateway that are created for this application.
+
+  Every rule contains attributes:
+  - `name`         - (`string`, required) Rule name.
+  - `priority`     - (`string`, required) Rule evaluation order can be dictated by specifying an integer value from 1 to 20000 with 1 being the highest priority and 20000 being the lowest priority.
+  - `backend`      - (`string`, optional) Backend settings` key
+  - `listener`     - (`string`, required) Listener's key
+  - `rewrite`      - (`string`, optional) Rewrite's key
+  - `url_path_map` - (`string`, optional) URL Path Map's key
+  - `redirect`     - (`string`, optional) Redirect's key
   EOF
-  default     = ["TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256", "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384", "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256", "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"]
-  type        = list(string)
+  type = map(object({
+    name         = string
+    priority     = number
+    backend      = optional(string)
```

Comment:
it's either backend or redirect, they are mutually exclusive, but at least one is required - I think we should have a validation block for that

Reply:
I added validation, but either `backend`, `redirect` or `url_path_map` is required, so it's a little bit extended.